server/src/main/java/org/elasticsearch/common/util/concurrent/ThreadContext.java

```diff
@@ -177,6 +177,28 @@ public StoredContext stashAndMergeHeaders(Map<String, String> headers) {
         return () -> threadLocal.set(context);
     }
 
+    /**
+     * Removes the current context and resets a new context that is a copy of the current one except that the request
+     * headers do not contain the given headers to remove. The removed context can be restored when closing the returned
+     * {@link StoredContext}.
+     * @param headersToRemove the request headers to remove
+     */
+    public StoredContext removeRequestHeaders(Set<String> headersToRemove) {
```

Comment:
nit: I'd call this `stashWithoutHeader()`

Reply:
As the other `stashXXX()` methods start off with the default thread context, I avoided prefixing this one with `stash`. I don't feel strongly on it though. What do you reckon @DaveCTurner ?

Reply:
I see, feel free to merge like it is so that we don't waste time on naming :)

Reply:
Yeah I also don't want to say `stash`. I didn't come up with an obviously better name, although maybe I'd have called it something like `withoutRequestHeaders` or `removingRequestHeaders` rather than using the imperative `remove`. NBD anyway, naming is hard.